Redistricting, state courts, and territories updates

ballotpedia opened the pull request:
combines the following pull requests and the requested changes to those:
https://github.com/opencivicdata/ocd-division-ids/pull/314 (us_state_courts.csv)

https://github.com/opencivicdata/ocd-division-ids/pull/313 (us_sldu.csv)

https://github.com/opencivicdata/ocd-division-ids/pull/312 (us_sldl.csv)

nuttingd1 reviewed (commented):
Overall I think this looks good, just have a few minor tweaks. Also, curious about the 'validTo' header, shouldn't it be 'validThrough' according to the [docs](https://github.com/opencivicdata/docs.opencivicdata.org/blob/master/proposals/0002.rst) @jloutsenhizer?

nuttingd1 commented on identifiers/country-us/census_autogenerated/us_sldu.csv:
```diff
@@ -1,1983 +1,2020 @@
+ocd-division/country:us/state:nc/sldu:28,North Carolina State Senate district 28,,
+ocd-division/country:us/state:nc/sldu:29,North Carolina State Senate district 29,,
+ocd-division/country:us/state:nc/sldu:30,North Carolina State Senate district 30,,
+ocd
```
doesn't look like the comment preview is showing the right lines, but referring to lines 1785-1787

jloutsenhizer replied:
@ballotpedia can you address this comment? I don't have a strong opinion either way

MiniCodeMonkey commented:
Would be great to get this merged in, so we can get the final OCD identifiers stabilized for the updated districts

MiniCodeMonkey commented:
It looks like districts were updated for the ND and SD house, but not for the senate (the districts are the same for the house and senate).

The following entries are thus missing:

```
ocd-division/country:us/state:nd/sldu:4a,North Dakota State Senate district 4A
ocd-division/country:us/state:nd/sldu:4b,North Dakota State Senate district 4B
ocd-division/country:us/state:nd/sldu:9a,North Dakota State Senate district 9A
ocd-division/country:us/state:nd/sldu:9b,North Dakota State Senate district 9B

ocd-division/country:us/state:sd/sldu:26a,South Dakota State Senate district 26A
ocd-division/country:us/state:sd/sldu:26b,South Dakota State Senate district 26B
ocd-division/country:us/state:sd/sldu:28a,South Dakota State Senate district 28A
ocd-division/country:us/state:sd/sldu:28b,South Dakota State Senate district 28B
```

ballotpedia commented:
> It looks like districts were updated for the ND and SD house, but not for the senate (the districts are the same for the house and senate).
> 
> The following entries are thus missing:
> 
> ```
> ocd-division/country:us/state:nd/sldu:4a,North Dakota State Senate district 4A
> ocd-division/country:us/state:nd/sldu:4b,North Dakota State Senate district 4B
> ocd-division/country:us/state:nd/sldu:9a,North Dakota State Senate district 9A
> ocd-division/country:us/state:nd/sldu:9b,North Dakota State Senate district 9B
> 
> ocd-division/country:us/state:sd/sldu:26a,South Dakota State Senate district 26A
> ocd-division/country:us/state:sd/sldu:26b,South Dakota State Senate district 26B
> ocd-division/country:us/state:sd/sldu:28a,South Dakota State Senate district 28A
> ocd-division/country:us/state:sd/sldu:28b,South Dakota State Senate district 28B
> ```

Thanks for flagging this. We double-checked some sources but do not believe the North Dakota and South Dakota senates split their districts in this way. Here are the 2023 Senate membership lists for each chamber, which do not indicate such splits:
https://www.ndlegis.gov/sites/default/files/resource/miscellaneous/2023senateroster.pdf
https://sdlegislature.gov/Legislators/Listing/68

MiniCodeMonkey commented:
@ballotpedia Thanks for double checking this! The sources I'm getting these from are:

**North Dakota**: https://ndlegis.gov/downloads/redistricting-2021/finalshape1282021.zip

**South Dakota**: The Find My Legislators 2021 adopted map: https://sdlegislature.gov/Legislators/Find (District 26A/B and 28A/B are visible).

I am not sure why there is mismatch with the legislator rooster.

ballotpedia commented:
> @ballotpedia Thanks for double checking this! The sources I'm getting these from are:
> 
> **North Dakota**: https://ndlegis.gov/downloads/redistricting-2021/finalshape1282021.zip
> 
> **South Dakota**: The Find My Legislators 2021 adopted map: https://sdlegislature.gov/Legislators/Find (District 26A/B and 28A/B are visible).
> 
> I am not sure why there is mismatch with the legislator rooster.

Thank you -- sorry for the delay, but just an update that we are looking more closely into this

ballotpedia commented:
> @ballotpedia Thanks for double checking this! The sources I'm getting these from are:
> 
> **North Dakota**: https://ndlegis.gov/downloads/redistricting-2021/finalshape1282021.zip
> 
> **South Dakota**: The Find My Legislators 2021 adopted map: https://sdlegislature.gov/Legislators/Find (District 26A/B and 28A/B are visible).
> 
> I am not sure why there is mismatch with the legislator rooster.

We have confirmed that only the lower chambers (House) for both of these states have these separate subdivisions.

MiniCodeMonkey commented:
@ballotpedia Thanks for confirming!

jloutsenhizer commented:
> Overall I think this looks good, just have a few minor tweaks. Also, curious about the 'validTo' header, shouldn't it be 'validThrough' according to the [docs](https://github.com/opencivicdata/docs.opencivicdata.org/blob/master/proposals/0002.rst) @jloutsenhizer?

@nuttingd1 you're right about `validThrough` instead of `validTo`. I may have contributed to some of the incorrect use of validTo in other PRs. I'll look into rectifying that. @ballotpedia can you update this PR to make use of `validThrough`

jloutsenhizer commented on identifiers/country-us/census_autogenerated/us_sldu.csv:
```diff
@@ -1,1983 +1,2020 @@
+ocd-division/country:us/state:ma/sldu:1st_suffolk,Massachusetts First Suffolk district,,
+ocd-division/country:us/state:ma/sldu:2nd_suffolk,Massachusetts Second Suffolk district,,
+ocd-division/country:us/state:ma/sldu:hampden,Massachusetts Hampden district,,
+ocd-division/country:us/state:ma/sldu:berkshire_hampshire_franklin_and_hampden,"Massachusetts Berkshire, Hampshire, Franklin & Hampden district",,2023-01-02
+ocd-division/country:us/state:ma/sldu:berkshire_hampden_franklin_and_hampshire,"Massachusetts Berkshire, Hampden, Franklin, and Hampshire district",2023-01-03,
```
@ballotpedia can you take a look into this?

jloutsenhizer reviewed (commented):
Overall this looks good to me. @ballotpedia can you take a look into the issues raised by @nuttingd1

ballotpedia commented:
Hi @jloutsenhizer , I have pushed a change that should take care of the issues mentioned above in the us_sldu.csv file.
- Changed "validTo" to "validThrough"
- Fixed the MA state senate validFrom and validThrough dates
- Replaced underscores with hyphens in some VT state senate OCD IDs

I also ran the compile script and tests. Please let me know if there is anything else!

nuttingd1 commented:
@ballotpedia can you also change the "validTo" header in the us_sldl.csv file to "validThrough" and run the compile script again?

ballotpedia commented:
@nuttingd1 done! How's it look now?

nuttingd1 reviewed (approved):
@ballotpedia these changes look good to me!

stwlam reviewed (approved):
This looks good to me, but the person from our group who possibly had merge rights has left. @jloutsenhizer?

jloutsenhizer commented:
Thanks everyone for contributing to reviewing this. Merged.
